src/object_properties.cpp

```diff
@@ -120,6 +127,13 @@ void ObjectProperties::serialize(std::ostream &os) const
 	os << serializeString16(damage_texture_modifier);
 	writeU8(os, shaded);
 	writeU8(os, show_on_minimap);
+	if (nametag_bgcolor) {
+		writeU8(os, 1);
+		writeARGB8(os, nametag_bgcolor.value());
+	} else {
+		writeU8(os, 0);
+		writeARGB8(os, {});
+	}
```

Comment:
it'd be possible to save one byte here by encoding `nullopt` as `0,1,1,1` and any alpha=0 color as `0,0,0,0`

alternatively: why write a color when it's known that there isn't one?